ttnn/cpp/ttnn/operations/moreh/moreh_sgd/device/moreh_sgd_device_operation.hpp

```diff
@@ -0,0 +1,87 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_sgd {
+struct MorehSgdOperation {
+    struct operation_attributes_t {
+        float lr;
+        float momentum;
+        float dampening;
+        float weight_decay;
+        bool nesterov;
+        bool momentum_initialized;
+        const MemoryConfig param_out_memory_config;
+        const MemoryConfig momentum_buffer_out_memory_config;
+        const DeviceComputeKernelConfig compute_kernel_config;
+    };
+
+    struct tensor_args_t {
+        const Tensor& param_in;
+        const Tensor& grad;
+        const std::optional<const Tensor> momentum_buffer_in;
```

Comment:
Use const std::optional<Tensor>&

Reply:
resolved in [a981edd](https://github.com/tenstorrent/tt-metal/pull/12378/commits/a981edd24c59337a422bc4c59ae88b14489c75a5)